commit 69fd06c6d7f45bd8741119e2f2e3940786be98b3
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Wed Feb 8 17:00:02 2017 -0500

    xprtrdma: Disable pad optimization by default
    
    commit c95a3c6b88658bcb8f77f85f31a0b9d9036e8016 upstream.
    
    Commit d5440e27d3e5 ("xprtrdma: Enable pad optimization") made the
    Linux client omit XDR round-up padding in normal Read and Write
    chunks so that the client doesn't have to register and invalidate
    3-byte memory regions that contain no real data.
    
    Unfortunately, my cheery 2014 assessment that this optimization "is
    supported now by both Linux and Solaris servers" was premature.
    We've found bugs in Solaris in this area since commit d5440e27d3e5
    ("xprtrdma: Enable pad optimization") was merged (SYMLINK is the
    main offender).
    
    So for maximum interoperability, I'm disabling this optimization
    again. If a CM private message is exchanged when connecting, the
    client recognizes that the server is Linux, and enables the
    optimization for that connection.
    
    Until now the Solaris server bugs did not impact common operations,
    and were thus largely benign. Soon, less capable devices on Linux
    NFS/RDMA clients will make use of Read chunks more often, and these
    Solaris bugs will prevent interoperation in more cases.
    
    Fixes: 677eb17e94ed ("xprtrdma: Fix XDR tail buffer marshalling")
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1351_69fd06c6d7f45bd8741119e2f2e3940786be98b3.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1351_69fd06c6d7f45bd8741119e2f2e3940786be98b3.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
